test/verify/check-networking-firewall

```diff
@@ -177,6 +178,66 @@ class TestFirewall(MachineCase):
             b.click("caption button.btn-primary")
             b.wait_present("#cockpit_modal_dialog li input[data-id='empty']")
 
+    @skipImage("Custom ports were introduced in 191", "rhel-7-6-distropkg")
+    def testAddCustomServices(self):
+        b = self.browser
+
+        self.login_and_go("/network/firewall")
+
+        # click on the "Add Services" button
+        b.wait_present("caption button.btn-primary:enabled")
+
+        def open_dialog():
+            b.click("caption button.btn-primary")
+            b.wait_present("#add-services-dialog input[value='ports']")
+            b.click("#add-services-dialog input[value='ports']")
+            b.wait_present("#tcp-ports")
+            b.wait_visible("#tcp-ports")
+
+            time.sleep(1) # We need some time before /etc/services is parsed
+
+        def set_field(sel, val, expected):
+            b.set_input_text(sel, val)
+            b.wait_val("#service-name", expected)
+
+        def check_error(text):
+            b.wait_present(".error-msg:contains({0})".format(text))
+            b.wait_visible(".error-msg:contains({0})".format(text))
+
+        def save(ajde, name, tcp, udp):
```

Comment:
"ajde"?

Reply:
Ah, yeah :smile: 
`id` is built-in function and I prefer not to use it for variables. And this was first thing that I thought of. I will update to something more sensible